xds/src/main/java/io/grpc/xds/FilterChainMatchingProtocolNegotiators.java

```diff
@@ -149,9 +150,8 @@ SelectedConfig select(InetSocketAddress localAddr, InetSocketAddress remoteAddr)
         filterChains = filterOnSourcePort(filterChains, remoteAddr.getPort());
 
         if (filterChains.size() > 1) {
-          log.log(Level.FINE, "Found more than one matching filter chains: {0}", filterChains);
-          throw new IllegalStateException("Found more than one matching filter chains.");
-          // TODO(chengyuanzhang): should we just return any matched one?
+          throw new IllegalStateException("Found more than one matching filter chains. This should "
+              + "not be possible as ClientXdsClient validated the chains for uniqueness.");
```

Comment:
> `ClientXdsClient`

do you mean xds in general, or specifically this network filter chain match?

Reply:
I mean specifically `ClientXdsClient.checkForUniqueness()`, but that's a private method.